draft up new popover component (#4117)
grommet/hpe-design-system: aries-site/src/pages/templates/popover.mdx

import { Example } from '../../layouts';
import { PopoverAnatomy, PopoverSimpleExample, PopoverInlineExample } from '../../examples';

> <Example

Reviewer: Should we include the code under these examples?

Author: yes will add thank you

<Example
  figma="https://www.figma.com/design/5uOrsL2qk8XwwH8C1ZpDb6/HPE-Design-System-Components?node-id=34141-2"
  width="100%"
>
  <PopoverInlineExample />
</Example>

## Anatomy

<Example caption="Diagram illustrating the 4 parts of a Popover." plain>
  <PopoverAnatomy />
</Example>

| Label  | Region                  | Purpose                               | Required | Notes                                               |
| :----: | ----------------------- | ------------------------------------- | :------: | --------------------------------------------------- |
| **1**  | **Title**               | Identifies the Popover.               | Optional |                                                     |
| **2**  | **Close Button**        | Closes the Popover.                   | Required | --                                                  |
| **3**  | **Body**                | Provides contextual information.      | Required | May include other elements that are not text.       |
| **4**  | **Footer**              | Contains additional Popover actions.  | Optional | May contain anchors.                                |

## Content
Popovers should be:

> - Relevant - The Popover should relate to its trigger and provide additional contextual information.
> - The content should not contain crucial information or actions required to complete a task.
> - Short -  Messages should be simple, too the point, and small to avoid overloading the user and taking up too much screen real estate. Follow the voice and tone guidelines for writing crisp and effective content.

Reviewer: ```suggestion
- Relevant - The Popover should relate to its trigger and provide additional contextual information.
- The content should not contain crucial information or actions required to complete a task.
- Short -  Messages should be simple, too the point, and small to avoid overloading the user and taking up too much screen real estate. Follow the voice and tone guidelines for writing crisp and effective content.
```

Formatting of the second bullet point doesn't match up with the other two

It is recommended that Popovers:
- Use Paragraph/Small and Anchor/Small to save real estate.
- Avoid using titles if it provides no additional context.

> ## Accessibility

Reviewer: In this section should we mention we don't recommend using headings for titles?

Author: yes we can add let me chat with Kenny

Author: added

| **4** | **Footer**    | Contains additional actions.          | Optional | May include anchors.                       |

## Content guidelines

Popovers should be:
Relevant: It should relate to its trigger.
Not crucial: It should not include crucial information or actions needed to complete a task.
Short: It should be concise to avoid overwhelming the user and taking up excessive screen space. Follow [voice and tone](https://design-system.hpe.design/foundation/voice-and-tone) guidelines for writing effective content.

Recommended practices:
- Use `Paragraph/Small` and `Anchor/Small` to save space.
- Avoid using titles if they do not provide additional context.
- Use `Text` for the title instead of a heading.

## Accessibility

Ensure accessibility by including an `a11yTitle` on:
- Triggers without labels, such as icon-only buttons.
- The Close button, describing the Popover's contents.

| Key   | Interaction                                                                                                      |
| ----- | ---------------------------------------------------------------------------------------------------------------- |

> | **Tab** | Highlights interactive elements. Focus remains inside the Popover until exited. Once exited, focus returns to the trigger. |

Reviewer 2: ```suggestion
| **Tab** | Highlights interactive elements. Focus remains within the Popover until it is closed. Once exited, focus returns to the trigger. |
```

import { Example } from '../../layouts';
import {
  PopoverAnatomy,
  PopoverSimpleExample,
  PopoverInlineExample,
  PopoverDoExample,
  PopoverDontPlacementExample,
  PopoverDontNestExample
} from '../../examples';

<Example

>   figma="https://www.figma.com/design/5uOrsL2qk8XwwH8C1ZpDb6/HPE-Design-System-Components?node-id=34141-2"

Author: yes I will update!

>   figma="https://www.figma.com/design/5uOrsL2qk8XwwH8C1ZpDb6/HPE-Design-System-Components?node-id=34141-2"

Author: @KennyAtHPE I need the updated link for this

  width="100%"
  code='https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/popover/PopoverSimpleExample.js'
>
  <PopoverSimpleExample />
</Example>

## Use cases
### Providing inline contextual help

A Popover can be triggered through an icon-only button to offer helpful contextual information in a non-intrusive manner.

> 

Author: fixed

<Example
  figma="https://www.figma.com/design/5uOrsL2qk8XwwH8C1ZpDb6/HPE-Design-System-Components?node-id=34141-2"
  width="100%"
  code='https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/popover/PopoverInlineExample.js'
>
  <PopoverInlineExample />
</Example>

## Anatomy

<Example caption="Diagram illustrating the 4 parts of a Popover." plain>
  <PopoverAnatomy />
</Example>

| Label | Region        | Purpose                              | Required | Notes                                      |
| :---: | ------------- | ------------------------------------ | :------: | ------------------------------------------ |
| **1** | **Title**     | Identifies the Popover.               | Optional |                                            |
| **2** | **Close Button** | Closes the Popover.                | Required |                                            |
| **3** | **Body**      | Provides contextual information.      | Required | May include non-text elements.             |
| **4** | **Footer**    | Contains additional actions.          | Optional | May include anchors.                       |

> ## Content guidelines

Reviewer 3: Suggestion to follow the same way of writing for all bullet points here: 

Popover's content should be:
- **Relevant:** It should relate to its trigger.
- **Not crucial:** It should not include crucial information or actions needed to complete a task.
- **Short:** It should be concise to avoid overwhelming the user and taking up excessive screen space. Follow [voice and tone](https://design-system.hpe.design/foundation/voice-and-tone) guidelines for writing effective content.

Reviewer 2: +1 looks like we lost the "bullet" points

Author: added bullets back!

import { Example } from '../../layouts';
import {
  PopoverAnatomy,
  PopoverSimpleExample,
  PopoverInlineExample,
} from '../../examples';

<Example
  figma="https://www.figma.com/design/5uOrsL2qk8XwwH8C1ZpDb6/HPE-Design-System-Components?node-id=34141-2"
  width="100%"
  code='https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/popover/PopoverSimpleExample.js'
>
  <PopoverSimpleExample />
</Example>

## Use cases
### Providing inline contextual help

A Popover can be triggered through an icon-only button to offer helpful contextual information in a non-intrusive manner.

<Example
  figma="https://www.figma.com/design/5uOrsL2qk8XwwH8C1ZpDb6/HPE-Design-System-Components?node-id=34141-2"
  width="100%"
  code='https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/popover/PopoverInlineExample.js'
>
  <PopoverInlineExample />
</Example>

## Anatomy

> <Example caption="Diagram illustrating the 4 parts of a Popover." plain>

Reviewer 4: ```suggestion
<Example caption="Diagram illustrating the 4 parts of a Popover which include an optional title, required close button, required body content, and an optional footer." plain>
```

  <PopoverAnatomy />
</Example>

| Label | Region        | Purpose                              | Required | Notes                                      |
| :---: | ------------- | ------------------------------------ | :------: | ------------------------------------------ |

> | **1** | **Title**     | Identifies the Popover.               | Optional |                                            |

Reviewer 4: Only use titlecase when referring to the component. Sentence case when referring to the "thing" or instance.

```suggestion
| **1** | **Title**     | Identifies the popover.               | Optional |                                            |
```

> | **2** | **Close Button** | Closes the Popover.                | Required |                                            |

Reviewer 4: I believe our convention is to use a center-aligned Checkmark icon with aria-label="required" for required regions.

Author: Ill double check

Author: I can add it will look like the following in Layer.
![Screenshot 2024-10-16 at 9 42 08 PM](https://github.com/user-attachments/assets/01eed140-b01f-4a5e-a494-5b85f06c1a2a)

> | **2** | **Close Button** | Closes the Popover.                | Required |                                            |

Reviewer 4: ```suggestion
| **2** | **Close button** | Closes the popover.                | Required |  Closed by mouse-click and/or keyboard. A popover also closes by any clicks outside of the popover. |
```

| **3** | **Body**      | Provides contextual information.      | Required | May include non-text elements.             |
| **4** | **Footer**    | Contains additional actions.          | Optional | May include anchors.                       |

> 

Reviewer 4: Are we planning to provide spacing annotations as well? Similar to https://design-system.hpe.design/components/layer#anatomy

Author: ![Screenshot 2024-10-16 at 9 48 37 PM](https://github.com/user-attachments/assets/e8963513-ac65-4bca-8426-dd182db8574d)
yes I have these then will add the table for when the screen is smaller is has the values

Reviewer 4: beautiful.

## Content guidelines

Popovers should be:

> - Relevant: It should relate to its trigger.

Reviewer 4: ```suggestion
- Relevant: Content should relate to its trigger.
```

> - Not crucial: It should not include crucial information or actions needed to complete a task.

Reviewer 4: ```suggestion
- Non-crucial: Contents should not include crucial information or actions required to complete a task.
```

> - Short: It should be concise to avoid overwhelming the user and taking up excessive screen space. Follow [voice and tone](https://design-system.hpe.design/foundation/voice-and-tone) guidelines for writing effective content.

Reviewer 4: ```suggestion
- Short: Content should be concise to avoid overwhelming the user and occupying excessive screen space. Follow [voice and tone](https://design-system.hpe.design/foundation/voice-and-tone) guidelines for writing effective content.
```

Recommended practices:
- Use `Paragraph/Small` and `Anchor/Small` to save space.
- Avoid using titles if they do not provide additional context.
- Use `Text` for the title instead of a heading.

## Accessibility

Ensure accessibility by including an `a11yTitle` on:
- Triggers without labels, such as icon-only buttons.

> - The Close button, describing the Popover's contents.

Reviewer 4: "describing the Popover's contents." -- confused by this. Can we just say "Close button"?

Add `aria-haspopup` as well as `aria-controls` on:
- The trigger in which opens the popover.
   - `aria-haspopup` tells users that there is a pop-up associated with the element.
   - `aria-controls` gives users more detailed information about which specific element will be affected when they interact with the element that has aria-haspopup.

> 

Reviewer 4: ```suggestion
Popover should be operable by both mouse and keyboard events.

Keyboard focus should be contained or "trapped" within the popover while it is open. 
```

> | Key   | Interaction                                                                                                      |

Reviewer 4: These are great.

| ----- | ---------------------------------------------------------------------------------------------------------------- |

> | **Tab** | Highlights interactive elements. Focus remains within the Popover until it is closed. Once exited, focus returns to the trigger. |

Reviewer 4: ```suggestion
| **Tab** | Highlights interactive elements. Focus remains within the popover until it is closed. Once exited, focus returns to the trigger. |
```

> | **Esc** | Exits the Popover at any time.                                                                                   |

Reviewer 4: ```suggestion
| **Esc** | Exits the popover at any time.                                                                                   |
```

| **2** | **Close Button** | Closes the Popover.                | Required |                                            |
| **3** | **Body**      | Provides contextual information.      | Required | May include non-text elements.             |
| **4** | **Footer**    | Contains additional actions.          | Optional | May include anchors.                       |

## Content guidelines

Popovers should be:
- Relevant: It should relate to its trigger.
- Not crucial: It should not include crucial information or actions needed to complete a task.
- Short: It should be concise to avoid overwhelming the user and taking up excessive screen space. Follow [voice and tone](https://design-system.hpe.design/foundation/voice-and-tone) guidelines for writing effective content.

Recommended practices:
- Use `Paragraph/Small` and `Anchor/Small` to save space.
- Avoid using titles if they do not provide additional context.
- Use `Text` for the title instead of a heading.

## Accessibility

Ensure accessibility by including an `a11yTitle` on:
- Triggers without labels, such as icon-only buttons.
- The Close button, describing the Popover's contents.

> Ensure to add `aria-haspopup` as well as `aria-controls` on:

Reviewer 2: ```suggestion
Add `aria-haspopup` as well as `aria-controls` on:
```

- The trigger in which opens the popover.

> 

Reviewer 2: ```suggestion
```

> - `aria-haspopup` tells users that there is a pop-up associated with the element.

Reviewer 2: ```suggestion
   - `aria-haspopup` tells users that there is a pop-up associated with the element.
```

Reviewer 2: Should these be sub-bullets under "The trigger which opens the popover"?

> - `aria-controls` gives users more detailed information about which specific element will be affected when they interact with the element that has aria-haspopup.

Reviewer 2: ```suggestion
   - `aria-controls` gives users more detailed information about which specific element will be affected when they interact with the element that has aria-haspopup.
```

      label: 7,
      property: 'elevation',
      value: 'medium',
    },
  ]}
/>

## Content guidelines

Popovers should be:
- Relevant: Content should relate to its trigger.
- Non-crucial: Contents should not include crucial information or actions required to complete a task.
- Short: Content should be concise to avoid overwhelming the user and occupying excessive screen space. Follow [voice and tone](https://design-system.hpe.design/foundation/voice-and-tone) guidelines for writing effective content.

Recommended practices:
- Use `Paragraph/Small` and `Anchor/Small` to save space.
- Avoid using titles if they do not provide additional context.
- Use `Text` for the title instead of a heading.

## Accessibility

Ensure accessibility by including an `a11yTitle` on:
- Triggers without labels, such as icon-only buttons.
- The Close button.

Add `aria-haspopup` as well as `aria-controls` on:
- The trigger in which opens the popover.
   - `aria-haspopup` tells users that there is a pop-up associated with the element.
   - `aria-controls` gives users more detailed information about which specific element will be affected when they interact with the element that has `aria-haspopup`.

> Popover should be operable by both mouse and keyboard events.

Reviewer 4: Need another space. This is getting added to the previous bulletpoint.

```suggestion

Popover should be operable by both mouse and keyboard events.
```
